docs/querying/api.md

```diff
@@ -1022,3 +1022,205 @@ $ curl -XPOST http://localhost:9090/api/v1/admin/tsdb/clean_tombstones
+        "range": {
+          "start": {
+            "line": 0,
+            "character": 4
+          },
+          "end": {
+            "line": 0,
+            "character": 6
+          }
+        },
+        "newText": "go_gc_duration_seconds_count"
+      }
+    }
+  ]
+```
+
+### Hover
+This endpoint returns a documentation according to the given word contains in the promQL expression provided
+
+```
+POST /api/v1/langserver/hover
+```
+
+Body format:
+
+```json
+{
+  "expr": "a PromQL expression",
+  "positionLine": 0 # The line (0 based) for which the metadata is queried.
+  "positionChar": 2 # The column (0 based) for which the metadata is queried. Characters are counted as UTF-16 code points.
+}
+```
+
+Example:
+```json
+$ curl -XPOST 'localhost:9090/api/v1/langserver/hover' -H "Content-Type: application/json" --data '{"expr": "sum(go_info)", "positionLine":0, "positionChar":10}'
+
+{
+  "contents":{
+    "kind": "markdown",
+    "value": "### go_info\n\n__Metric Help:__ Information about the Go environment.\n\n__Metric Type:__  gauge\n\n\n\n__PromQL Type:__ instant vector\n\n"
+  },
+  "range": {
+    "start": {
+      "line": 0,
+      "character": 4
+    },
+    "end": {
+      "line": 0, 
+      "character": 11
+    }
+  }
+}
+```
+
+```json
+$ curl -XPOST 'localhost:9090/api/v1/langserver/hover' -H "Content-Type: application/json" --data '{"expr": "sum(go_info)", "positionLine":0, "positionChar":3}'
+
+{
+  "contents": {
+    "kind": "markdown",
+    "value": "## `\u003caggregation\u003e_over_time()`\n\nThe following functions allow aggregating each series of a given range vector\nover time and return an instant vector with per-series aggregation results:\n\n* `avg_over_time(range-vector)`: the average value of all points in the specified interval.\n* `min_over_time(range-vector)`: the minimum value of all points in the specified interval.\n* `max_over_time(range-vector)`: the maximum value of all points in the specified interval.\n* `sum_over_time(range-vector)`: the sum of all values in the specified interval.\n* `count_over_time(range-vector)`: the count of all values in the specified interval.\n* `quantile_over_time(scalar, range-vector)`: the ¤å-quantile (0 Ôëñ ¤å Ôëñ 1) of the values in the specified interval.\n* `stddev_over_time(range-vector)`: the population standard deviation of the values in the specified interval.\n* `stdvar_over_time(range-vector)`: the population standard variance of the values in the specified interval.\n\nNote that all values in the specified interval have the same weight in the\naggregation even if the values are not equally spaced throughout the interval.\n\n\n__PromQL Type:__ instant vector\n\n"
+  },
+  "range": {
+    "start": {
+      "line": 0,
+      "character": 0
+    },
+    "end": {
+      "line": 0,
+      "character": 26
+    }
+  }
+}
+```
+
+### Signature Help
+This endpoint returns the signature associated to the method selected in the promQL expression provided
```

Comment:
```suggestion
This endpoint provides a function signature if the provided cursor position is within a PromQL function call. Otherwise a falsy value is returned.

The result is a  [`SignatureHelp`](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_signatureHelp).
```

Reply:
done